### ISSUE TITLE
How do i run adb terminal

### ISSUE THREAD

**Issue by foodmaher** [CypherpunkArmory/UserLAnd#693]
# Can I run adb commands on UserLAnd

Do I need root permission to run ADB commands for control my device by Userland?

I haven't tried anything yet because I do not know if it will require root permission and , do will it work on Userland?

# Describe how your task behaves in a regular Linux environment

As you know it behaves normally and has been tested by me on a Windows 10 operating system.

**Comment by lewisdonofrio** [comment]
I was on my way down this road..... I installed my setup like this http://www.tinyurl.com/donofriodexnote8 and I have adb installed in my ubuntu 18.04 would like to see if it network wise can connect because we're never going to get access to physical hardware (usb cables and the like) I want to code on this setup and I've yet to get time....

**Comment by MatthewTighe** [comment]
Are you asking about either:
1. Running ADB inside UserLAnd?
2. Running ADB from a computer to access UserLAnd files?

For 1:
I don't think you would need to be root to run ADB from inside UserLAnd if you could figure out how to get it installed. Normally ADB detects connections over USB however, which would not make sense in referencing the device it is running on. You also won't have access to USB drivers anyway I believe.

For 2:
You still don't need to root your device, but I believe ADB works best when APKs are installed by hand. This allows you to run the following commands:
```adb shell
run-as tech.ula```
which you can use to access your UserLAnd files

**Comment by foodmaher** [comment]
@MatthewTighe 

**You say** that **ADB** detects the connection via a **USB** cable only, but is it possible to connect to it by creating a *"connection service"** as an example? You have linked the **Nox Player** emulator with ADB so that it can be known as the **Android studio** for running application.

**Comment by corbinlc** [comment]
Q: You say that ADB detects the connection via a USB cable only, but is it possible to connect to it by creating a connection service as an example? 
A: We don't know how possible that is.  

If I understand correctly Android Studio does not support arm, but I wouldn't be surprised if you could get the platform-tools, like ADB, working.

**Comment by EnderNightLord-ChromeBook** [comment]
> Q: You say that ADB detects the connection via a USB cable only, but is it possible to connect to it by creating a connection service as an example?
> A: We don't know how possible that is.
> 
> If I understand correctly Android Studio does not support arm, but I wouldn't be surprised if you could get the platform-tools, like ADB, working.

Yeah... It could be possible.

**Comment by mpottinger** [comment]
@corbinlc Yes i believe I have had adb working in the past. The way to do it is to enable wireless adb, and then you can connect via localhost in userland 127.0.0.1.

Still it is a pain because you need to connect the phone to PC first to enable it.

**Comment by Urahara** [comment]
Any update on this? I trying this too but without success.

**Comment by virtimus** [comment]
Hi there.
Also spent some time on this - I think option 1 would be nice feature.
Installation of adb in UL/Ubuntu is quite simple:

```
sudo apt install android-sdk-platform-tools
adp devices
``` -> empty list

Then connected the phone to PC (standard ubuntu with adb installed) and switched phone to tcp debug mode:

`adb tcpip 6555` (default port was causing problems)

disconnected phone

back to UL:
`adb connect localhost:6555`

displays "connected to localhost:6555" but adb server crashes

tried with ip:
`adb connect 192.168.2.131:6555`

_adb connect 192.168.2.131:6555
* daemon not running; starting now at tcp:5037
* daemon started successfully
connected to 192.168.2.131:6555_

and if You try:

`adb shell`

_* daemon not running; starting now at tcp:5037
* daemon started successfully
error: no devices/emulators found_

**Comment by corbinlc** [comment]
Closing as part of a repository cleanup. If this issue is still relevant, feel free to reopen it or file a fresh one.
